components/mpas-ocean/src/shared/mpas_ocn_tracer_nonlocalflux.F

```diff
@@ -58,6 +61,85 @@ module ocn_tracer_nonlocalflux
 
 contains
 
+!***********************************************************************
+!
+!  routine ocn_brine_rejction_tend
+!
+!> \brief   Computes tendency term due to brine rejection
+!> \author  Luke Van Roekel
+!> \date    11/08/24
+!> \details
+!>  This routine computes the tendency for salinity from brine rejection, based on 
+!>      Nguyen et al 2009
+!
+!-----------------------------------------------------------------------
+
+   subroutine ocn_brine_rejection_tendency(tend, layerThickness, seaIceSalinityFlux, index_salinity, dt, err)
+      !----------------------------------------------------------------
+      !
+      ! variables
+      !
+      !----------------------------------------------------------------
+
+      real (kind=RKIND), dimension(:,:,:), intent(inout) :: &
+         tend !< tracer tendency, only will modify the salinity tend here
+
+      real (kind=RKIND), dimension(:), intent(in) :: &
+         seaIceSalinityFlux
+
+      real (kind=RKIND), dimension(:,:), intent(in) :: &
+         layerThickness
+
+      real, dimension(:), allocatable :: sis_flux
+
+      real (kind=RKIND) :: dt
+
+      integer, intent(in) :: index_salinity
+
+      integer :: err
+
+      !----------------------------------------------------------------
+      !
+      ! local variables
+      !
+      !----------------------------------------------------------------
+
+      integer :: iCell, k, nCells
+
+      real (kind=RKIND) :: remainder, z, integral, D, A, rejectedSalt
+
+      nCells = nCellsHalo(1)
+
+      allocate(sis_flux(nVertLevels+1))
+
+      do iCell = 1, nCells
+        if (seaIceSalinityFlux(iCell) > 0) then !only compute for salinity into ocean
+           !this is set to 1 such that across the BLD we don't have a huge salt flux divergence.
+           sis_flux(:) = 1.0_RKIND*seaIceSalinityFlux(iCell)
+           D = boundaryLayerDepth(iCell)
+           z = 0.0_RKIND
+           A = (config_brine_param_n + 1) / D**(config_brine_param_n+1)
+           k = minLevelCell(iCell)
+           ! need a salinity flux top and bottom and the flux into a cell is the divergence??
+           do while (z < D)
+              sis_flux(k) = min(1.0_RKIND,A*z**config_brine_param_n)*seaIceSalinityFlux(iCell)
```

Comment:
```suggestion
              sis_flux(k) = max(0.0_RKIND,1.0_RKIND - A*z**config_brine_param_n)*-seaIceSalinityFlux(iCell)
```
When I scoped this out in python, a few other small changes were required to make your approach and mine equivalent (happy to share code), but the basic idea is to have `sis_flux` at the top of the model equal `seaIceSalinityFlux` and `sis_flux` below the BL equal 0. I think this is more intuitive because in its current form, it suggests that salt is fluxing from the bottom up rather than the top down, even though the divergence from both ways of writing it is identical.

Reply:
Note: in Fortran `0**0 = 1` for most compilers so there would be an issue here if users set `config_brine_param_n = 0`. Probably the easiest thing to do is just set the value for z=0 outside this loop.